toolchain/parse/tree_and_subtrees.h

```diff
@@ -16,6 +17,12 @@ namespace Carbon::Parse {
 // This requires a complete tree.
 class TreeAndSubtrees {
  public:
+  // A range of tokens, returned by GetSubtreeTokenRange.
+  struct TokenRange {
+    Lex::TokenIndex start;
+    Lex::TokenIndex end;
```

Comment:
I would use either:
```suggestion
    Lex::TokenIndex begin;
    Lex::TokenIndex end;
```
Or:
```suggestion
    Lex::TokenIndex start;
    Lex::TokenIndex stop;
```